src/main/java/org/jboss/pnc/common/json/moduleprovider/BAConfigProvider.java

```diff
@@ -0,0 +1,53 @@
+package org.jboss.pnc.common.json.moduleprovider;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.jboss.pnc.buildagent.moduleconfig.BuildAgentModuleConfig;
+import org.jboss.pnc.common.json.AbstractModuleConfig;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.jsontype.NamedType;
+
+public class BAConfigProvider <T extends AbstractModuleConfig> implements ConfigProvider<T> {
+    
+    private List<ProviderNameType<T>> moduleConfigs;
+    private Class<T> type;
+    
+    public BAConfigProvider(Class<T> type) {
+      this.type = type;  
+      moduleConfigs = new ArrayList<>();  
+      moduleConfigs.add(new ProviderNameType(BuildAgentModuleConfig.class,"build-agent-config"));
```

Comment:
The linkage between class (BuildAgentModuleConfig.class) and module name ("build-agent-config") seems the only module specific bits in this class. Can't we move other methods to an abstract class ?